v0/destinations/engage/transform.js

```diff
@@ -0,0 +1,261 @@
+const { set } = require("lodash");
+const {
+  defaultRequestConfig,
+  constructPayload,
+  removeUndefinedAndNullValues,
+  getFieldValueFromMessage,
+  simpleProcessRouterDest,
+  defaultPutRequestConfig,
+  defaultDeleteRequestConfig
+} = require("../../util");
+const { TRANSFORMER_METRIC } = require("../../util/constant");
+const ErrorBuilder = require("../../util/error");
+const { DESTINATION } = require("./config");
+const { getDestinationExternalID } = require("../../util");
+
+const { EventType } = require("../../../constants");
+const { mappingConfig, ConfigCategories } = require("./config");
+const {
+  refinePayload,
+  getUID,
+  generatePageName,
+  getLists
+} = require("./utils");
+
+const responseBuilder = (payload, endpoint, method, Config) => {
+  if (!payload) {
+    throw new ErrorBuilder()
+      .setMessage("Parameters could not be found.")
+      .setStatus(400)
+      .setStatTags({
+        destType: DESTINATION,
+        stage: TRANSFORMER_METRIC.TRANSFORMER_STAGE.TRANSFORM,
+        scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.SCOPE,
+        meta: TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.META.BAD_PARAM
+      })
+      .build();
+  }
+  const { publicKey, privateKey } = Config;
+  const response = defaultRequestConfig();
+  // Authentication Ref : https://engage.so/docs/api/
+  const basicAuth = Buffer.from(`${publicKey}:${privateKey}`).toString(
+    "base64"
+  );
+  response.headers = {
+    "Content-Type": "application/json",
+    Authorization: `Basic ${basicAuth}`
+  };
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+  response.endpoint = endpoint;
+  response.method = method;
+  return response;
+};
+
+// Engage Api Doc for identify, track, page calls Ref:https://engage.so/docs/api/users
+
+const identifyResponseBuilder = (message, Config) => {
+  const payload = constructPayload(
+    message,
+    mappingConfig[ConfigCategories.IDENTIFY.name]
+  );
+  const { traits } = message;
+  const uid = getUID(message);
```

Comment:
If uId is not sent, can we make Identify call? If not we need to handle it here.

Reply:
as we are using the update or create API we can't make the call. 
To handle it we can either have throw error or use the create API which will work without userId as well.